### PR TITLE
Run component test in parallel

### PR DESCRIPTION
**Pull request by amitkrout**
## What is the purpose of this change? What does it change?

To reduce test time both locally and on CI too.
## Was the change discussed in an issue?
fixes - part of #1473

## How to test changes?

make test-cmp-e2e && make test-cmp-sub-e2e

### CODE REVIEW

**Comment by amitkrout** [comment]
@kadel @mohammedzee1000 Pr related to parallel run keeps growing. Can you guys start reviewing from this component/sub-component pr cc_ @cdrage

File: Makefile [hunk 1]
```diff
@@ -114,12 +114,12 @@ test-json-format-output:
 # Run component e2e tests
 .PHONY: test-cmp-e2e
 test-cmp-e2e:
-	go test -v github.com/openshift/odo/tests/integration --ginkgo.focus="odoCmpE2e" -ginkgo.slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -ginkgo.v -timeout $(TIMEOUT)
+	ginkgo -v -nodes=$(TEST_EXEC_NODES) -focus="odoCmpE2e" slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -randomizeAllSpecs  tests/integration/ -timeout $(TIMEOUT)
```
Comment (cdrage) [inline Makefile]: why are we now randomizing? you need to document this in your PR description / commit.

File: docs/development.adoc [hunk 2]
```diff
@@ -158,34 +158,25 @@ $ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--ena
...
-.Procedure:
-
-To deploy an integration test:
+.How to write:
 
+Refer to the odo clean test link:https://github.com/openshift/odo/blob/master/tests/template/template_cleantest_test.go[`template`].
 
-* For the entire integration suite use:
-----
-$ make test-e2e
```
Comment (cdrage) [inline docs/development.adoc]: @amitkrout again, you're removing documentation and not adding it again... no documentation on `make test-integration`

Comment (amitkrout) [reply docs/development.adoc]: Ahh...my bad. Will update it

File: Makefile (hunk 1 above)
Comment (amitkrout) [reply Makefile]: https://github.com/openshift/odo/pull/1797/files#diff-b67911656ef5d18c4ae36cb6741b7965R114

File: docs/development.adoc (hunk 2 above)
Comment (amitkrout) [reply docs/development.adoc]: https://github.com/openshift/odo/pull/1797/files#diff-c26dca7a0015d5429db3d986bc160252R183

**Pull request merged**